src/elements/Badge/Badge.tsx

```diff
@@ -44,7 +90,19 @@ const Badge: FC<BadgeProps> = ({ children, status, icon }: BadgeProps) => {
                 )}
                 {icon && <span data-test-id="badge-icon">{cloneElement(icon, { size: IconSize.Size12 })}</span>}
                 <span className="text-center text-xxs font-sans font-normal">{children}</span>
-            </span>
+                {onDismiss && (
+                    <button
+                        data-test-id="badge-icon"
+                        className={dismissClasses[style]}
+                        onClick={(event) => {
+                            event.stopPropagation();
+                            onDismiss();
+                        }}
+                    >
+                        <Reject size={IconSize.Size12} />
```

Comment:
Shouldn't the icon disappear as soon as `onDismiss` is called? If the Badge works as a kind of selectable tag, this would be the expected behaviour in my opinion.

Reply:
since I do not know what the use case of the dismiss "button" is or how the badge is going to be used and also the design just states that it is possible to add a dismiss cross I would suggest to leave it like that for the moment.